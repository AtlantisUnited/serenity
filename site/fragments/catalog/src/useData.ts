import { useQuery } from '@apollo/client'

import gql          from 'graphql-tag'

export const query = gql`
  query Categorys {
    categoryGroups {
      rows {
        id
        name
        children {
          id
          name
        }
      }
    }
  }
`

export const useData = () => {
  const { data } = useQuery(query)

  return data && data.categoryGroups ? data.categoryGroups.rows : []
}
