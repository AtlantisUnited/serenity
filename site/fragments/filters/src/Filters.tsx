import React                   from 'react'

import { Collapse, Panel }     from '@ui/collapse'
import { CheckIcon }           from '@ui/icons'
import { Box, Column, Layout } from '@ui/layout'
import { Text }                from '@ui/text'

export const Filters = ({
  data = [],
  check,
  title,
  activeCategory,
  selectCategory,
  activeKey,
  onChange,
}) => (
  <Column>
    <Layout flexBasis={48} alignItems='center'>
      <Text fontWeight='medium' fontSize='extra' color='black' lineHeight='extra'>
        {title}
      </Text>
    </Layout>
    <Layout flexBasis={20} />
    <Box border='black' borderRadius='extra' boxSizing='border-box'>
      <Collapse activeKey={activeKey} onChange={onChange}>
        {data.map((item: any) => (
          <Panel
            header={item.name}
            key={item.id}
            check={
              check && (
                <CheckIcon
                  width={16}
                  height={16}
                  color={
                    item.children.some((child) => child.id === activeCategory)
                      ? 'black'
                      : 'rgb(153, 153, 153)'
                  }
                />
              )
            }
          >
            <Column>
              {(item as any).children.map((category: any, num: any) => (
                <Box key={category.id} mt={num === 0 ? 0 : 12} alignItems='center'>
                  <Layout>
                    {check && (
                      <CheckIcon
                        width={16}
                        height={16}
                        color={activeCategory === category.id ? 'black' : 'rgb(153, 153, 153)'}
                      />
                    )}
                  </Layout>
                  <Layout flexBasis={16} flexShrink={0} />
                  <Layout>
                    <Text
                      onClick={() => selectCategory(category.id)}
                      color='stormdust'
                      cursor='pointer'
                    >
                      {category.name}
                    </Text>
                  </Layout>
                </Box>
              ))}
            </Column>
          </Panel>
        ))}
      </Collapse>
    </Box>
  </Column>
)
