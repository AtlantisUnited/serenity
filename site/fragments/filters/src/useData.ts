import gql          from 'graphql-tag'

import { useQuery } from '@apollo/client' // eslint-disable-line import/no-extraneous-dependencies

export const query = gql`
  query Categorys {
    categoryGroups {
      rows {
        id
        name
        children {
          id
          name
        }
      }
    }
  }
`

export const useData = () => {
  const { data } = useQuery(query)

  return data && data.categoryGroups ? data.categoryGroups.rows : []
}
