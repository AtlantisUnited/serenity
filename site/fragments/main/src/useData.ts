import gql          from 'graphql-tag'

import { useQuery } from '@apollo/client'

export const query = gql`
  query Categories {
    categories {
      rows {
        id
        name
      }
    }
  }
`

export const useData = () => {
  const { data } = useQuery(query)

  return data && data.categories ? data.categories.rows : []
}
