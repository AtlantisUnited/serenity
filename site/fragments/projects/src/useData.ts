import gql          from 'graphql-tag'

import { useQuery } from '@apollo/client'

export const query = gql`
  query SearchProjects($filters: SearchProjectsFilter, $query: String!) {
    search {
      projects(filters: $filters, query: $query) {
        rows {
          id
          name
          category {
            id
            name
          }
          description
          budget
          legalEntitiesOnly
          views
          replyCount
          publicationDate
        }
      }
    }
  }
`

export const useData = (searchQuery, searchCategory) => {
  const { data = {} }: any = useQuery(query, {
    variables: {
      query: searchQuery,
      filters: searchCategory
        ? {
            categoryId: searchCategory,
          }
        : null,
    },
  })

  return (data && data.search && data.search.projects && data.search.projects.rows) || []
}
