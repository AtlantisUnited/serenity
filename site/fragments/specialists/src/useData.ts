import gql          from 'graphql-tag'

import { useQuery } from '@apollo/client'

export const query = gql`
  query SearchSpecialists($filters: SearchSpecialistsFilter, $query: String!) {
    search {
      specialists(filters: $filters, query: $query) {
        rows {
          id
          specialisation {
            main {
              name
            }
            additional {
              name
            }
          }
          description
          interaction {
            formOfWork
            name
          }
          profile {
            photo {
              url
            }
            personalInformation {
              firstName
              lastName
            }
          }
          portfolio {
            name
            images {
              id
              url
            }
          }
          rating
        }
      }
    }
  }
`

export const useData = (searchQuery, searchCategory) => {
  const { data = {} }: any = useQuery(query, {
    variables: {
      query: searchQuery,
      filters: searchCategory
        ? {
            specialisationId: searchCategory,
          }
        : null,
    },
  })

  return (data && data.search && data.search.specialists && data.search.specialists.rows) || []
}
