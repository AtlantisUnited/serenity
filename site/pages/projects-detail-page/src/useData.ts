import gql          from 'graphql-tag'

import { useQuery } from '@apollo/client'

export const queryProjects = gql`
  query Projects($filters: ProjectsFilter) {
    projects(filters: $filters) {
      rows {
        name
        status
        category {
          name
        }
        description
        address {
          formatted
        }
        beginningOfWork
        photos {
          id
          url
        }
        publicationDate
        budget
        worksheet
        legalEntitiesOnly
        views
        replyCount
        owner {
          member {
            openProjects
            completedProjects
          }
          activity {
            last
          }
          profile {
            personalInformation {
              firstName
              lastName
            }
            photo {
              url
            }
            address {
              formatted
            }
          }
        }
      }
    }
  }
`

export const queryReplies = gql`
  query Me($projectId: ID) {
    me {
      profile {
        type
        photo {
          url
        }
      }
      replies(projectId: $projectId) {
        id
        status
        discussion {
          id
          messages {
            id
            content
            author {
              photo {
                url
              }
              personalInformation {
                firstName
                lastName
              }
            }
            member {
              id
              rating
              interaction {
                formOfWork
                name
              }
              specialisation {
                main {
                  name
                }
                additional {
                  name
                }
              }
            }
          }
        }
      }
    }
  }
`

export const useDataProjects = (id) => {
  const { data } = useQuery(queryProjects, {
    variables: {
      filters: {
        id,
      },
    },
  })

  return (data && data.projects && data.projects.rows[0]) || {}
}

export const useDataReplies = (id) => {
  const { data } = useQuery(queryReplies, {
    variables: {
      projectId: id,
    },
  })

  return (data && data.me && { profile: data.me.profile, replies: data.me.replies }) || {}
}
