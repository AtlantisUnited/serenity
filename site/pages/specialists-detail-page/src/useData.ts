import gql          from 'graphql-tag'

import { useQuery } from '@apollo/client'

export const query = gql`
  query Specialists($filters: SpecialistsFilter) {
    specialists(filters: $filters) {
      rows {
        id
        reviewCount
        completedProjects
        specialisation {
          main {
            id
            name
          }
          additional {
            id
            name
          }
        }
        description
        interaction {
          numberOfEmployees
          formOfWork
          name
        }
        profile {
          photo {
            url
          }
          personalInformation {
            firstName
            lastName
          }
          contactInformation {
            phone {
              number
            }
          }
          address {
            formatted
          }
          website
        }
        portfolio {
          name
          images {
            id
            url
          }
        }
        rating
        reviews {
          id
          rating
          comment
          customer {
            personalInformation {
              firstName
              lastName
            }
            photo {
              url
            }
          }
        }
      }
    }
  }
`

export const useData = (id) => {
  const { data } = useQuery(query, {
    variables: {
      filters: {
        id,
      },
    },
  })

  return (data && data.specialists && data.specialists.rows[0]) || {}
}
