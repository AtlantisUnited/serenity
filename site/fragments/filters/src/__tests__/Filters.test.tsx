import React              from 'react'

import { MockedProvider } from '@apollo/react-testing'
import { ThemeProvider }  from '@ui/theme'
import { render }         from '@testing-library/react'

import Filters            from '..'
import { query }          from '../useData'

const mocks = [
  {
    request: { query },
    result: {
      data: {
        categoryGroups: {
          rows: [
            {
              id: 1,
              name: 'name',
              children: [
                {
                  id: 1,
                  name: 'name',
                },
              ],
            },
          ],
        },
      },
    },
  },
]

describe('Filters fragment', () => {
  describe('snapshots', () => {
    it('should match latest render snapshot', () => {
      const { asFragment } = render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider>
            <Filters />
          </ThemeProvider>
        </MockedProvider>
      )

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
