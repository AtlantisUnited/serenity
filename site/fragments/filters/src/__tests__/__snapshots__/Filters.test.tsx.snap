// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters fragment snapshots should match latest render snapshot 1`] = `
<DocumentFragment>
  .emotion-3 {
  background-color: #ffffff;
  border-radius: 3px;
  width: 100%;
  font-family: Inter;
  word-break: break-word;
  outline: none;
}

.emotion-3 .rc-collapse-anim-active {
  -webkit-transition: height 0.2s ease-out;
  transition: height 0.2s ease-out;
}

.emotion-3 > .rc-collapse-item {
  border-top: 1px solid #d9d9d9;
}

.emotion-3 > .rc-collapse-item:first-of-type {
  border-top: none;
}

.emotion-3 > .rc-collapse-item > .rc-collapse-header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  line-height: 24px;
  font-size: 15px;
  font-weight: 500;
  padding: 16px 56px 16px 24px;
  color: #666;
  outline: none;
  cursor: pointer;
}

.emotion-3 > .rc-collapse-item > .rc-collapse-header .arrow {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  vertical-align: middle;
  right: 24px;
  top: 20px;
}

.emotion-3 > .rc-collapse-item > .rc-collapse-header .rc-collapse-extra {
  padding-top: 4px;
  margin-right: 12px;
  line-height: normal;
}

.emotion-3 > .rc-collapse-item-active > .rc-collapse-header {
  color: #000;
}

.emotion-3 > .rc-collapse-item-disabled > .rc-collapse-header {
  cursor: not-allowed;
  color: #999;
}

.emotion-3 .rc-collapse-content {
  overflow: hidden;
  color: #666;
  box-sizing: border-box;
}

.emotion-3 .rc-collapse-content > .rc-collapse-content-box {
  font-size: 13px;
  line-height: 20px;
  padding: 6px 24px;
}

.emotion-3 .rc-collapse-content > .rc-collapse-content-box:first-of-type {
  padding-top: 0px;
}

.emotion-3 .rc-collapse-content > .rc-collapse-content-box:last-of-type {
  padding-bottom: 16px;
}

.emotion-3 .rc-collapse-content-inactive {
  display: none;
}

.emotion-5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 48px;
  -ms-flex-preferred-size: 48px;
  flex-basis: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  color: rgb(0,0,0);
  font-weight: 500;
  font-size: 18px;
  line-height: 1.3;
  font-family: Inter;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 20px;
  -ms-flex-preferred-size: 20px;
  flex-basis: 20px;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 1px solid rgba(0,0,0,0.08);
  border-radius: 4px;
}

<div
    class="emotion-5"
    display="flex"
    width="100%"
  >
    <div
      class="emotion-1"
      display="flex"
    >
      <span
        class="emotion-0"
        color="black"
        font-family="primary"
        font-size="extra"
        font-weight="medium"
      />
    </div>
    <div
      class="emotion-2"
      display="flex"
    />
    <div
      class="emotion-4"
      display="flex"
    >
      <div
        class="rc-collapse emotion-3"
      />
    </div>
  </div>
</DocumentFragment>
`;
